what4/src/What4/Expr/Builder.hs

```diff
@@ -946,6 +954,340 @@ evalBoundVars sym e vars exprs = do
                             }
   evalBoundVars' tbls sym e
 
+
+newtype ExprTransformer t (tp1 :: BaseType) (tp2 :: BaseType) a =
```

Comment:
Most of the code added here in `What4.Expr.Builder` is undocumented, and as a result, I can't really follow what is going on here. Some additional comments would definitely be appreciated. (I'll trust your judgment to determine how much documentation is appropriate.)